apple/internal/apple_framework_import.bzl

```diff
@@ -116,13 +118,18 @@ def _apple_framework_import_impl(ctx):
             objc_provider_fields["weak_sdk_framework"] = depset(ctx.attr.weak_sdk_frameworks)
         objc_provider_fields["static_framework_file"] = framework_imports_set
 
+    providers.append(
```

Comment:
```
provider_fields = {}
if transitive_sets:
    provider_fields["framework_imports"] = depset(transitive = transitive_sets)
providers.append(
    AppleFrameworkImportInfo(**provider_fields)
)
```

this way, there's no memory overhead on empty depsets when depending on a bunch of static `apple_framework_import` targets

Reply:
done